lib/std/crypto/tls/Client.zig

```diff
@@ -468,7 +468,7 @@ pub fn init(stream: anytype, ca_bundle: Certificate.Bundle, host: []const u8) In
                         read_seq += 1;
                         P.AEAD.decrypt(cleartext, ciphertext, auth_tag, record_header, nonce, p.server_handshake_key) catch
                             return error.TlsBadRecordMac;
-                        break :c cleartext;
+                        break :c cleartext[0..std.mem.trimRight(u8, cleartext, "\x00").len];
```

Comment:
You could simply use `std.mem.trimRight(u8, cleartext, "\x00");` here, too!

Reply:
It is later used as `[]u8` and trimRight returns `[]const u8` which forces this ugly construction.

With `break :c std.mem.trimRight(u8, cleartext, "\x00");`:
```
$ zig run --zig-lib-dir ../zig/lib example/http_get_std.zig -- tutanota.com
/home/ianic/Code/zig/lib/std/crypto/tls/Client.zig:478:63: error: expected type '[]u8', found '[]const u8'
                var ctd = tls.Decoder.fromTheirSlice(cleartext[0 .. cleartext.len - 1]);
                                                     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/ianic/Code/zig/lib/std/crypto/tls/Client.zig:478:63: note: cast discards const qualifier
/home/ianic/Code/zig/lib/std/crypto/tls.zig:453:32: note: parameter type declared here
    pub fn fromTheirSlice(buf: []u8) Decoder {
                               ^~~~
referenced by:
    connectTcp: /home/ianic/Code/zig/lib/std/http/Client.zig:1357:60
    connect: /home/ianic/Code/zig/lib/std/http/Client.zig:1492:38
    5 reference(s) hidden; use '-freference-trace=7' to see all references
```

Reply:
you can `@constCast`

Reply:
Thanks!